crates/sui-graphql-rpc/src/server/builder.rs

```diff
@@ -442,6 +491,36 @@ async fn get_or_init_server_start_time() -> &'static Instant {
     ONCE.get_or_init(|| async move { Instant::now() }).await
 }
 
+/// Starts an infinite loop that periodically updates the `AvailableRange`.
+pub(crate) async fn update_available_range(
+    db: &Db,
+    available_range: Arc<RwLock<AvailableRange>>,
```

Comment:
An `Arc<RwLock<AvailableRange>>` is overkill here, and indeed undesirable: Under high load, the incoming requests reading the available range will starve the writer thread and the range never gets updated. You just need an `AtomicU64` to store the `last` part of the available range, and I think it's even fine for this to have a `Relaxed` order on stores and loads because you are caching it once per request in the context.